book/notes/liquidation-math.md

```diff
@@ -0,0 +1,108 @@
+# Margin Account Liquidation Math
+
+This is a short note to explain the math behind margin account liquidation. Dango's margin account liquidation is based on Euler's dynamic liquidation bonus method, or "Dutch Liquidation". Please see this [paper](https://docs.euler.finance/Dutch_Liquidation_Analysis.pdf) for more details.
+
+## Defintions
+
+* Collateral power: The adjustment factor for the collateral value of a given collateral token. Meaning, if the collateral token has a collateral power of 0.9, then the value of the collateral token is 90% of its actual value.
+* Utilization rate: The ratio of the total value of the debts to the total value of the account's collateral adjusted for the collateral power. When the utilization rate is greater than 1, the account is undercollateralized and can be liquidated.
+* Health factor: The inverse of the utilization rate. When the health factor is less than 1, the account is undercollateralized and can be liquidated.
+* Liquidation bonus: The amount of extra collateral that is awarded to the liquidator to incentivize them to liquidate the account.
+
+## Calculating The Maximum Repayable Debt
+
+When liquidating the account, we set a target utilization rate (which is a fraction between 0 and 1) that we want to reach after the liquidation. We then calculate the maximum amount of debt that can be repaid to reach this target utilization rate.
+
+In our system, the liquidation bonus is defined as a function of the health factor:
+
+$$
+\mathrm{liquidation\ bonus} = 1 - \mathrm{health\ factor}
+$$
+
+This means that the more undercollateralized the account is, the higher the liquidation bonus.
+
+Consider the below graph showing an undercollateralized margin account:
+
+![undercollateralized margin account](./liquidation-graph.png)
+
+The x-axis represents the value of account's debt and the y-axis represents the value of account's collateral adjusted for the collateral power. The point $p(t_0)$ is the account's current status. Since it is below the orange line, it is undercollateralized and eligible for liquidation. The blue line represents the trajectory of the liquidation. The black line represents all points at which the account's utilization rate is equal to the target utilization rate. The maximum repayable debt therefore is found at the intersection of the blue line and the black line.
+
+### Liquidation Trajectory
+
+We can think of a user’s position at time t as a point p(t) = (d(t), c(t)) in the debt and adjusted collateral plane. To derive the equation for the liquidation trajectory, we can use the point slope form of the equation for a line:
```

Comment:
Using "time" here is confusing because we don't use timestamps any where in the math. It's more confusing because later in the text you also use $t$ to denote the target health rate, which is unrelated to time. Let's just use $(c_0, d_0)$ to denote the current status of the position.

Reply:
In the below equation to define the gradient I use t0 and t1 to show the point before and after liquidation. But can remove it here.

Reply:
Changed to d vs d_1 and c vs c_1